mesa/space.py

```diff
@@ -962,15 +963,12 @@ def get_heading(
         Args:
             pos_1, pos_2: Coordinate tuples for both points.
         """
-        one = np.array(pos_1)
-        two = np.array(pos_2)
+        heading = np.asarray(pos_2) - np.asarray(pos_1)
+
         if self.torus:
-            one = (one - self.center) % self.size
-            two = (two - self.center) % self.size
-        heading = two - one
-        if isinstance(pos_1, tuple):
-            heading = tuple(heading)
-        return heading
+            heading = (heading + self.half_size) % self.half_size
```

Comment:
Does this mean `heading` is always < `self.half_size` for both axes? It doesn't seem to be the case for the old version.

Reply:
Good catch, I missed the calculation order so I simplified `self.size - self.half_size` to `self.half_size`, but that shouldn't be the case, because of course the modulo is done first and only after that the subtraction.